test/cypress/integration/amount/amount.cy.test.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+const productStubs = require('../../stubs/products-stubs')
+const serviceStubs = require('../../stubs/service-stubs')
+
+const gatewayAccountId = 666
+const productExternalId = 'a-product-id'
+
+describe('Amount page', () => {
+  describe('when the Payment Link has no price', () => {
+    beforeEach(() => {
+      cy.task('setupStubs', [
+        productStubs.getProductByExternalIdStub({
+          external_id: productExternalId,
+          reference_enabled: true,
+          reference_label: 'invoice number',
+          type: 'ADHOC'
+        }),
+        serviceStubs.getServiceSuccess({
+          gatewayAccountId: gatewayAccountId,
+          serviceName: {
+            en: 'Test service name'
+          }
+        })
+      ])
+
+      Cypress.Cookies.preserveOnce('session')
+    })
+
+    it('should redirect to the `Amount` page', () => {
+      cy.visit('/pay/a-product-id/amount')
+
+      cy.get('[data-cy=back-link]').should('have.attr', 'href', '/pay/a-product-id/reference')
```

Comment:
Could do something like: 
```
cy.get('a').invoke('attr', 'href').should('eq', '/pay/a-product-id/reference').then(href => {
    cy
      .request(href)
      .its('status')
      .should('eq', 200)
})
```
to validate the link works as well.

Reply:
Good point - in this case, it probably is not worth doing.
- We can keep the test smaller and easier to understand.
- We will be creating another test for the main `product` page later.